### PR TITLE
ParlayVault feature

### PR DESCRIPTION
**Pull request by hinkypunks**

### CODE REVIEW

**Review by danijelthales — DISMISSED** [review]

File: contracts/SportMarkets/Parlay/ParlayMarketsAMM.sol [hunk 1]
```diff
@@ -366,7 +371,7 @@ contract ParlayMarketsAMM is Initializable, ProxyOwned, ProxyPausable, ProxyReen
             _positions,
             _sUSDPaid
         );
-        uint safeBoxAmount = ((_sUSDPaid - sUSDAfterFees) * safeBoxImpact) / (safeBoxImpact + parlayAmmFee);
+        uint safeBoxAmount = _getSafeBoxAmount(_sUSDPaid, sUSDAfterFees, msg.sender);
```
Comment (kirilaa) [inline contracts/SportMarkets/Parlay/ParlayMarketsAMM.sol:374]: for future consistency and not braking the code:
`_getSafeBoxAmount(_sUSDPaid, sUSDAfterFees, _differentRecipient);`

with the audit merge
`_differentRecepient => _differentRecipient`

**Comment by danijelthales** [comment]
Please add a new variable that will control how many times the same market can be used on tickets in the round (e.g. no more than 4 tickets in a round can contain same market)

File: contracts/OvertimeVaults/ParlayVault.sol [hunk 2]
```diff
@@ -93,6 +94,9 @@ contract ParlayVault is Initializable, ProxyOwned, PausableUpgradeable, ProxyRee
 
     mapping(uint => uint) public allocationSpentInARound;
 
+    mapping(uint => mapping(address => uint)) public marketNumberPerRound;
+    uint public maxMarketNumberPerRound;
```
Comment (danijelthales) [inline contracts/OvertimeVaults/ParlayVault.sol:98]: Would rename to maxMarketUsedInRoundCount

File: contracts/OvertimeVaults/ParlayVault.sol [hunk 3]
```diff
@@ -93,6 +94,9 @@ contract ParlayVault is Initializable, ProxyOwned, PausableUpgradeable, ProxyRee
 
     mapping(uint => uint) public allocationSpentInARound;
 
+    mapping(uint => mapping(address => uint)) public marketNumberPerRound;
```
Comment (danijelthales) [inline contracts/OvertimeVaults/ParlayVault.sol:97]: would rename to marketUsedInRoundCount

**Review by danijelthales — CHANGES_REQUESTED** [review]

**Review by danijelthales — APPROVED** [review]

**Pull request merged**